infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogEntrySinkBufferManager.java

```diff
@@ -84,8 +84,7 @@ public void processBuffer() {
             LogReplicationEntryMetadataMsg metadata = entry.getMetadata();
             if (metadata.getTimestamp() <= lastProcessedSeq) {
                 buffer.remove(metadata.getPreviousTimestamp());
-            } else if (metadata.getPreviousTimestamp() <= lastProcessedSeq) {
-                sinkManager.processMessage(entry);
+            } else if (metadata.getPreviousTimestamp() <= lastProcessedSeq && sinkManager.processMessage(entry)) {
```

Comment:
Even if the message is not processed could it still be acknowledged? as the ack is not dependent on this output? Could it be that in one of the failing cases we still ACK?

Reply:
We will send ACK msgs, since that is not dependent on the msg processed..But, the timestamp in the ACK will be the lastProcessedTs, which is updated only when the msg is processed.

> if (shouldAck()) {
>             LogReplicationEntryMetadataMsg metadata = generateAckMetadata(dataMessage);
>             log.trace("Sending an ACK {}", metadata);
>             return getLrEntryAckMsg(metadata);
>         }

The generateAckMetadata() generates an ACK msg and sets the ACK-timestamp to lastProcessedTs.

So if some msgs do not get processed at Standby, we don't update the lastProcessedTs, and the ACKs sent will never correspond to the msg ignored.

Reply:
Ok, yes you're right I had not looked into the method. One thing is, If you realize the message is malformed, as in the case the message is not processed we're taking the metadata of the last received entry meaning that some other metadata like prevTs (for instance) does not match the 'lastProcessedTs'. From a quick look I think for now as that field is not used we're fine. If you can add this comment as a TODO so we fix it in the future to avoid other bugs derived from this.